src/lib.rs

```diff
@@ -571,6 +571,37 @@ impl CecLogicalAddresses {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum TryFromCecLogicalAddressesError {
+    UnknownPrimaryAddress,
+    InvalidPrimaryAddress,
+}
+
+impl TryFrom<cec_logical_addresses> for CecLogicalAddresses {
+    type Error = TryFromCecLogicalAddressesError;
+    fn try_from(addresses: cec_logical_addresses) -> Result<Self, Self::Error> {
+        let primary = CecLogicalAddress::try_from(addresses.primary)
+            .map_err(|_| TryFromCecLogicalAddressesError::UnknownPrimaryAddress)?;
+        let primary = KnownCecLogicalAddress::new(primary)
+            .ok_or(TryFromCecLogicalAddressesError::InvalidPrimaryAddress)?;
+
+        let addresses = HashSet::from_iter(addresses.addresses.into_iter().enumerate().filter_map(
+            |(addr, mask)| {
+                let addr = addr as i32;
+                if mask == 1 {
+                    KnownAndRegisteredCecLogicalAddress::new(
+                        CecLogicalAddress::try_from(addr).ok()?,
+                    )
+                } else {
+                    None
+                }
```

Comment:
Readability suggestion

```suggestion
            |(logicalAddr, addrIsSetMask)| {
                // addresses.addresses is an mask array indicating logical addresses
                // if addresses.addresses[idx] is non-zero, it means that logical address with value
                // index idx is in the group    
                let logicalAddr = logicalAddr as cec_logical_address;
                if addrIsSetMask != 0 {
                    KnownAndRegisteredCecLogicalAddress::new(
                        CecLogicalAddress::try_from(logicalAddr).ok()?,
                    )
                } else {
                    None
                }
```

Reply:
Updated, but with style some style differences.

Ideally this should be in the `libcec-sys` crate documentation, but that looks like it's generated by bindgen. :(